src/clinical/clinical-service.ts

```diff
@@ -231,8 +232,31 @@ export const getPaginatedClinicalData = async (programId: string, query: Clinica
 	// Get all donors + records for given entity
 	const { donors, totalDonors } = await donorDao.findByPaginatedProgramId(programId, query);
 
+	const donorIds = donors.map((donor) => donor.donorId);
+
+	const isDefaultDonorSort = query.sort.includes('completionStats.coreCompletionPercentage');
+	const isInvalidSort = query.sort.includes('schemaMetadata.isValid');
+
+	const clinicalErrors = isInvalidSort
+		? (await getClinicalErrors(programId, donorIds)).clinicalErrors
+		: [];
```

Comment:
Because Clinical Entity Data is stored on Donors, and Error records are stored on Migrations and require Dictionary/Exception validation, we have to compare Entity Records against Error Records to insure proper sorting

This duplicates the work of the Clinical Errors resolver when querying for both Clinical Data and Clinical Errors
This makes the whole operation fairly slow for large programs

I'm not sure how to optimize this without a substantial refactor. 
Open to suggestions, but improvements should be made a separate ticket.

Reply:
It requires refactoring how the data is stored. Thanks for finding a way to work with it as is and highlighting the issue.

Reply:
Additional thought: this will be very quick for programs that don't have migration errors, so as long as we can keep migration errors in check its a non-issue.